kotlin/attribution-view-change/README.md

```diff
@@ -1,25 +1,31 @@
-# Attribution View Change
+# Attribution view change
 
-This sample demonstrates showing UI components responding to attribution view.
+Make UI elements respond to changes in the attribution view.
 
-![Attribution View Change App](attribution-view-change.png)
+![Image of attribution view change](attribution-view-change.png)
 
-## Features
+## Use case
 
-* ArcGISMap
-* Basemap
+ArcGIS Online basemaps, Esri data services, or Esri API technology used in an application require inclusion of an Esri attribution. Depending on its content, the attribution bar may be expandible and collapsible. In this case, expanding the attribution bar may obstruct some of the user interface in the application. To avoid this, the developer may want to make some of the UI elements reposition when the attribution bar is interacted with.
 
-## Developer Pattern
+## How to use the sample
 
-Set the bottom margin of FAB to respect the height of the attribution bar.
+Select the attribution bar to expand it, and observe how the floating action button moves up. Select the attribution bar again to minimize it and see how the floating action button moves down.
 
-```kotlin
- // set attribution bar listener
- val params = fab.layoutParams as CoordinatorLayout.LayoutParams
- mapView.addAttributionViewLayoutChangeListener { view, left, top, right, bottom, oldLeft, oldTop, oldRight, oldBottom ->
-     val heightDelta = bottom - oldBottom
-     params.bottomMargin += heightDelta
-     Toast.makeText(this, "new bounds [" + left + "," + top + "," + right + "," + bottom + "]" +
-             " old bounds [" + oldLeft + "," + oldTop + "," + oldRight + "," + oldBottom + "]", Toast.LENGTH_SHORT).show()
-        }
-```
+## How it works
+
+1. Add a layout change listener to the Map View with `addAttributionViewLayoutChangeListener`.
+2. Get the difference in pixels by using `hightDelta = oldBottom - bottom`.
+3. Change the floating action button's bottom margin using its `layoutParams`.
```

Comment:
this step is different to the Java Android implementation which was:
> Move the floating action button along the `MapView` with `animate().translationYBy(heightDelta)`.

I think I prefer the Java Android description.

Reply:
I was a bit confused because the implementation was different. The kotlin sample does not use `animate()` as far as I can tell

Reply:
ah yes, thanks for pointing that out. how about something like this?

> Move the floating action button along the MapView by using Android's `CoordinateLayout.LayoutParams` and setting it to `heightDelta`.

Reply:
sounds good!